docs/realtime.rst

```diff
@@ -0,0 +1,117 @@
+The Hypothesis Real Time API
+============================
+
+.. warning::
+
+   This document describes an API that is in the early stages of being
+   documented and refined for public use. Details may change, and your systems
+   may break in the future.
+
+In addition to the `HTTP API <http://h.readthedocs.io/en/latest/api/>`_
+Hypothesis has a WebSocket-based API that allows developers to receive near
+real-time notifications of annotation events.
+
+Overview
+--------
+
+.. note::
+
+   At the moment, use of the Real Time API is limited to server-side clients.
+   JavaScript applications running in the browser **will not be able to use this
+   API** due to security restrictions. We hope to lift this restriction in the
+   future.
+
+To use the Real Time API, you should open a WebSocket connection to the
+following endpoint::
+
+    wss://hypothes.is/ws
+
+Communication with this endpoint consists of JSON-encoded messages sent from
+client to server and vice versa.
+
+Server messages
+---------------
+
+Each messages from the server will be either an :term:`event` or a
+:term:`reply`:
+
+.. glossary::
+
+   event
+      An event is sent to clients as a result of an action taken elsewhere in
+      the system. For example: if an annotation is made which matches one of the
+      client's subscriptions, the client will receive an event message. All
+      event messages have a ``type`` field.
+
+   reply
+      A reply is sent in response to a message sent by the client. All replies
+      have an ``ok`` field which indicates whether the server successfully
```

Comment:
Can you add documentation for the format of errors here as well? It looks like we have a standard format for including errors in responses.